packages/stack/src/views/Stack/CardStack.tsx

```diff
@@ -275,21 +276,30 @@ export default class CardStack extends React.Component<Props, State> {
           : DefaultTransition;
 
       const {
+        animation,
         animationEnabled = Platform.OS !== 'web' &&
           Platform.OS !== 'windows' &&
           Platform.OS !== 'macos',
         gestureEnabled = Platform.OS === 'ios' && animationEnabled,
         gestureDirection = defaultTransitionPreset.gestureDirection,
         transitionSpec = defaultTransitionPreset.transitionSpec,
+        headerStyleInterpolator = defaultTransitionPreset.headerStyleInterpolator,
+      } = optionsForTransitionConfig;
+
+      let {
         cardStyleInterpolator = animationEnabled === false
           ? forNoAnimationCard
           : defaultTransitionPreset.cardStyleInterpolator,
-        headerStyleInterpolator = defaultTransitionPreset.headerStyleInterpolator,
         cardOverlayEnabled = (Platform.OS !== 'ios' &&
           optionsForTransitionConfig.presentation !== 'transparentModal') ||
           getIsModalPresentation(cardStyleInterpolator),
       } = optionsForTransitionConfig;
 
+      if (animation != null) {
+        cardStyleInterpolator =
+          AnimationTransitions[animation].cardStyleInterpolator;
+      }
```

Comment:
Does `animation` only affect `cardStyleInterpolator`? It doesn't seem enough. Specifying an animation should also configure the correct gesture direction, animation curves etc. depending on what's appropriate for the animation. It's essentially a replacement for a transition preset, not only `cardStyleInterpolator`.

The `animation` option shouldn't override `cardStyleInterpolator`. It should be the other way around. More specific options override less specific ones, so user should be able to specify `animation` and override part of that configuration such as transition spec, card style interpolator etc.